.github/workflows/coverage.yml

```diff
@@ -17,9 +17,34 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  check-file-change:
+    runs-on: ubuntu-latest
+    # see https://github.com/orgs/community/discussions/25722
+    if: ${{ github.event_name == 'push' || !github.event.pull_request.draft }}
+    outputs:
+      parachain_src: ${{ steps.filter.outputs.parachain_src }}
+      tee_src: ${{ steps.filter.outputs.tee_src }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      # Checks to see if any files in the PR/commit match one of the listed file types.
+      # We can use this filter to decide whether or not to build docker images
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: .github/file-filter.yml
+          list-files: shell
+
   test:
     name: Coverage Report
     runs-on: [ubuntu-latest]
+    needs:
+      - check-file-change
+    if: >
+      needs.check-file-change.outputs.parachain_src == 'true' ||
```

Comment:
I think it only covers the parachain pallets, maybe checking parachain part is enough

Reply:
Ah, yes. You are right. I was confused by the low coverage numbers. Because of quite many new pallets introduced.
BTW, we need to add more test code to have a better coverage.